instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/integration/sessionreplay/SessionReplaySegmentUtils.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sdk.integration.sessionreplay
+
+import com.datadog.android.sdk.rules.HandledRequest
+import com.google.gson.JsonElement
+import com.google.gson.JsonParser
+import okio.Buffer
+import java.io.ByteArrayOutputStream
+import java.util.Locale
+import java.util.zip.Inflater
+
+/**
+ * Utility functions for parsing and decompressing Session Replay segments in integration tests.
+ */
+internal object SessionReplaySegmentUtils {
+
+    private val SEGMENT_FORM_DATA_REGEX =
+        Regex("content-disposition: form-data; name=\"segment\"; filename=\"(.+)\"")
+    private val CONTENT_LENGTH_REGEX =
+        Regex("content-length: (\\d+)")
+
+    /**
+     * Extracts and parses a Session Replay segment from a handled request.
+     * The segment is decompressed and parsed as JSON.
+     *
+     * @return The parsed JSON element, or null if no segment found or parsing failed
+     */
+    fun HandledRequest.extractSrSegmentAsJson(): JsonElement? {
+        val compressedSegmentBody = resolveSrSegmentBodyFromRequest(requestBuffer.clone())
+        if (compressedSegmentBody.isNotEmpty()) {
+            // decompress the segment binary
+            return JsonParser.parseString(String(decompressBytes(compressedSegmentBody)))
+        }
+
+        return null
+    }
+
+    /**
+     * Resolves the Session Replay segment body from a multipart form request buffer.
+     *
+     * Example of a multipart form segment body:
+     * ```
+     * Content-Disposition: form-data; name="segment"; filename="db081a08-96ab-4931-a98e-b2dd2d9c1b34"
+     * Content-Type: application/octet-stream
+     * Content-Length: 1060
+     *
+     * <compressed segment as byte array of 1060 length>
+     * ```
+     *
+     * @param buffer The request buffer to parse
+     * @return The compressed segment bytes, or empty array if not found
+     */
+    private fun resolveSrSegmentBodyFromRequest(buffer: Buffer): ByteArray {
+        var line = buffer.readUtf8Line()
+        while (line != null) {
+            if (line.lowercase(Locale.ENGLISH).matches(SEGMENT_FORM_DATA_REGEX)) {
```

Comment:
just for the consistency in such calls

```suggestion
            if (line.lowercase(Locale.US).matches(SEGMENT_FORM_DATA_REGEX)) {
```